compiler/passes/initializerRules.cpp

```diff
@@ -295,29 +306,32 @@ void phase1Analysis(BlockStmt* body, AggregateType* t) {
 
     // Additionally, perform the following actions:
     // - add initialization for omitted fields
-    if (curField != NULL && (isInit == DID_NOT_FIND_INIT)) {
+    if (curField != NULL && (isInit == FOUND_NONE)) {
       // still have phase 1 statements and fields left to traverse
 
       if (BlockStmt* block = toBlockStmt(curExpr)) {
         if (block->isLoopStmt()) {
           // Special handling for loops.
-          bool foundInit = loopAnalysis(block, curField, seenField, &index, t);
+          bool foundInit = loopAnalysis(block,
+                                        curField,
+                                        seenField,
+                                        &index,
+                                        at);
+
           if (foundInit) {
-            // If the init call was in the loop body we just checked, any
-            // further action is incorrect (including the insertion of leftover
-            // omitted fields, since those fields would be inserted into the
-            // loop body and we don't allow that), so just exit the loop through
-            // the body and allow normal clean up to occur.
+            // If the init call was in the loop body we just checked,
+            // any further action is incorrect (including the insertion
+            // of leftover omitted fields, since those fields would be
+            // inserted into the loop body and we don't allow that),
+            // so just exit the loop through the body and allow normal
+            // clean up to occur.
             break;
           }
           curExpr = getNextStmt(curExpr, body, false);
           isInit = getInitCall(curExpr);
           continue;
         }
-      }/* else if (CondStmt* cond = toCondStmt(curExpr)) {
-        // Special handling for conditionals.
-        }*/
-
+      }
```

Comment:
I would like this to stay until we have the conditional statement support, please